### ISSUE TITLE
Exporting marker ranges as separate videos results in corrupt/broken videos

### ISSUE THREAD

**Issue by myonlinebusiness** [mltframework/shotcut#1597]
I have a relatively simple project with just one video track one audio track and subtitles burned in on the output. I have set three marker ranges on the timeline (_Marker 1, Marker 2, Marker 3_). Each marked section is about 40-41 seconds long. 
See screenshot from Shotcut main screen: ![shotcut-clip-overview](https://github.com/user-attachments/assets/012c40a6-177e-4f1b-beb4-b4c795a5baec)

Now I want to export these marker ranges individually as separate videos. For "Marker 1" (green) everything works fine.
However when exporting "Marker 2" (red) and "Marker 3" (blue) the exported video files seem to be corrupt/strange somehow.
Various video players show a length of over a minute (2 Min 3 sec for the video from "Marker 3" and 1 minute 22 seconds for the video from "Marker 2"). The videos play fine till the end of 40/41 seconds. Then, depending on the video player the timeline gets stuck or the player aborts as there is no more content beyond that point. However the timeline still goes till the end. I also noticed, that the exported subtitles track (not the burned in one, that shows fine in the video) does not contain anything, probably because the timestamps are screwed up as well.
See screenshot of vlc player: ![vlc-player-at-end-of-clip](https://github.com/user-attachments/assets/43b07690-7960-4c5e-a2ad-39c30488ef84)

To me (in laymans terms) it looks like the end timestamps of the exported videos are set incorrectly to the end position of that marker on the total track (1min 22sec for "Marker 2" and 2min 03sec for "Marker 3") instead of the length of the exported marker range.

Shotcut Version: 24.09.13
OS: Ubuntu 24.04.1 LTS

**Comment by ddennedy** [comment]
I was not able to reproduce this using the portable/AppImage/Snap build (they are the same) on Pop!_OS 22.04. I tested playback in VLC, Celluloid (mpv), and GNOME Videos. I tried using the same version as you as well as the Shotcut 24.10 beta. 

Are you using the export defaults (I did with and without hardware encoding)?

Did you try it more than once?  Reproduce it by testing the export before closing Shotcut. Then, for the bad video, right-click the export job, choose **View Log** and copy and paste it here.

How did you get and install Shotcut?

**Comment by myonlinebusiness** [comment]
Shotcut is installed via the default Snap package manager from Ubuntu. It seems to use the stable version "latest/stable:    24.09.13 2024-09-13 (1584)" from here: https://snapcraft.io/shotcut 

I did not use hardware encoding. Settings were as far as I know standard. I actually had the parallel processing originally on. But as I read that this sometimes causes trouble I deactivated the parallel processing. But the result is always the same corrupt video.
I had already done several exports initially, before creating this bug report to verify it's reproducible. 

Just tried again Exporting "Marker 3" in a freshly restarted Shotcut and it is (again) the same corrupt result. Attached is the export log from this last export job:
[job-log-marker-3.txt](https://github.com/user-attachments/files/17525856/job-log-marker-3.txt)
I did another export directly afterwards (without restarting Shotcut) and again same (bad) result.

**Comment by ddennedy** [comment]
Thanks for the additional info. It is weird that I could not reproduce it. The Snap is our build that I mentioned I was using. 

Meanwhile, I can suggest a workaround. Export using an intermediate preset such as DNxHR HQ (it will be a large file depending on duration). Then, do File > Close, double-click the export job to open the result in the **Source** player, select a portion of time using in and out points, add it to the playlist using drag-n-drop or the + button (Shift+A), and repeat for each additional sub-clip. Finally, Export > From > Each Playlist Item will do a batch export.

**Comment by ddennedy** [comment]
Some additional info you can provide that might help.
Open the problem video in Shotcut. What duration does it show, and does it play OK?
With it still open, choose Properties > menu > More Information. Copy and paste here.
Consider sharing one of your problem video exports here, if not directly, then a cloud drive link.

**Comment by myonlinebusiness** [comment]
I uploaded the striped down project and render result here: https://kdrive.infomaniak.com/app/share/768113/aeb5c9ea-8d54-46ba-817c-480c5623c0b2

sample.mlt is the project file
sample.mp4 is my last export.

Interesting thing is: When I open the rendered video in Shotcut it shows the (expected) duration of around 40 seconds. But every other video player (like vlc, gnome video etc.) shows the wrong length.

**Comment by myonlinebusiness** [comment]
Attached is the detail output from Shotcut when opening one of the exported (bad) videos:
[broken-exported-video-opened-in-shotcut.txt](https://github.com/user-attachments/files/17526261/broken-exported-video-opened-in-shotcut.txt)

**Comment by ddennedy** [comment]
I found the problem. It is the embedded subtitles (timed text track). That feature (on by default when you add subtitles) is not compatible with exporting with a range marker. The simpler workaround then is to **Export > Advanced > Other > Disable subtitles**. This only disables the embedding, not the rendering with the Subtitle Burn In. I do not think you need to embed them in the export as timed text since you are rendering them.

**Comment by bmatherly** [comment]
Fixed via https://github.com/mltframework/mlt/commit/335d0093dc5220466f49f7d307dca7bfc9d08e32
